React 18 support

Hi,

`govuk-react-jsx@7.0.1` has [a peer dependency on `"react": ">=16 <=17"`](https://unpkg.com/browse/govuk-react-jsx@7.0.1/package.json).

If React 18 works, it'd be good to get that updated to include version 18.

If React 18 doesn't work, it'd be good to get it supported.

Note that installing the package currently fails in a new project created with Create React App as recent versions of npm validate peer dependencies:

```
16:10 $ npm install govuk-react-jsx
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: govuk-react-jsx-playground@0.1.0
npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   react@"^18.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@">=16 <=17" from govuk-react-jsx@7.0.1
npm ERR! node_modules/govuk-react-jsx
npm ERR!   govuk-react-jsx@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,8 @@
 
 #### Features
 
+- React 18 support (Simply by loosening the peer dependency requirement).
+
 #### Breaking changes
 
 ---

--- a/package.json
+++ b/package.json
@@ -50,11 +50,11 @@
   "homepage": "https://github.com/surevine/govuk-react-jsx#readme",
   "dependencies": {
     "govuk-frontend": "4.0.1",
-    "react": "^17.0.1",
-    "react-dom": "^17.0.1",
+    "react": "^18.1.0",
+    "react-dom": "^18.1.0",
     "react-helmet": "^6.1.0",
-    "react-router": "^5.2.0",
-    "react-router-dom": "^5.2.0",
+    "react-router": "^6.3.0",
+    "react-router-dom": "^6.3.0",
     "yargs": "^16.2.0"
   },
   "devDependencies": {
@@ -69,7 +69,7 @@
     "@storybook/addon-actions": "^6.1.17",
     "@storybook/addons": "^6.1.17",
     "@storybook/react": "^6.1.17",
-    "@testing-library/react": "^11.2.5",
+    "@testing-library/react": "^13.2.0",
     "babel-eslint": "^10.1.0",
     "babel-loader": "^8.2.2",
     "babel-plugin-file-loader": "^2.0.0",

--- a/scripts/package.json
+++ b/scripts/package.json
@@ -23,9 +23,9 @@
   "homepage": "https://github.com/surevine/govuk-react-jsx#readme",
   "peerDependencies": {
     "govuk-frontend": "4.0.1",
-    "react-router-dom": "5",
-    "react-router": "5",
-    "react": ">=16 <=17",
+    "react-router-dom": ">=5",
+    "react-router": ">=5",
+    "react": ">=16",
     "react-helmet": "6"
   }
 }

